client/shared/src/test/scala/org/http4s/client/ClientRouteTestBattery.scala

```diff
@@ -129,6 +129,51 @@ abstract class ClientRouteTestBattery(name: String) extends Http4sSuite with Htt
     }
   }
 
+  GetRoutes.getPaths.get(GetRoutes.SimplePath).foreach { expected =>
+    val path = GetRoutes.SimplePath
+
+    test(s"$name Execute GET withEntity $path") {
+      serverClient().flatMap { case (server, client) =>
+        val address = server().addresses.head
+        val req =
+          Request[IO](uri = Uri.fromString(s"http://$address$path").yolo).withEntity("entity")
+        client()
+          .run(req)
+          .use(resp => expected.flatMap(checkResponse(resp, _)))
+          .assert
+      }
+    }
+
+    // test(s"$name Execute GET withEntity then withHeaders $path") {
```

Comment:
This currently fails for ember.

I think there are 2 issues here:
1. `withHeaders` currently removes internally added headers (like in this test content length that's added as part of `.withEntity("entity")` ). I _think_ these internally added headers should be left alone unless they are part of `withHeaders`.
2. Requests without content length confuse ember. This looks like the request counterpart of https://github.com/http4s/http4s/pull/5881

Reply:
Guessing that https://github.com/http4s/http4s/issues/6106 and https://github.com/http4s/http4s/issues/6107 will also be resolved if 1. was resolved.